### ISSUE TITLE
Wrong calculation of Isoweek for year 2015 (in tz != "UTC")

### ISSUE THREAD

**Issue by MoebiusAV** [Rdatatable/data.table#2407]
As posted in [https://stackoverflow.com/questions/43944430/r-why-does-the-default-package-for-isoweek-change-depending-how-i-use-it/46655291#46655291](url) data.table isoweek function sometimes gives a different result of ISO standart.

 I found an explination, running data.table function step by stepThe first step of function is converting Date as.POSIXlt. 
The result will be different if you input a character or a Date 
`as.POSIXlt('2015-01-02')`
`[1] "2015-01-02 COT"`

`as.POSIXlt(as.Date('2015-01-02'))`
`[1] "2015-01-02 UTC"`

As you can see, what change is the time Zone. 
In the third step of function, in both cases, it returns a Date in Local time
`(year_start <- as.POSIXct(paste0(as.POSIXlt(nearest_thurs)$year +1900L, "-01-01"))) `
`[1] "2015-01-01 COT"`
That way, in the forth and last step, the date difference between UTC and Local time gives some decimals 'aditional' to time difference between Local Time dates (in my case 5 hours, or 0.208). That is why the final result of weeks changes.

In conclusion, if you force date as character, the function will calculate everything in Local time and the result will be right
-- | --

**Comment by MichaelChirico** [comment]
Following up on my SO comment -- without a reproducible example, I can't confirm this works, but it seems the answer is to change:

https://github.com/Rdatatable/data.table/blob/master/R/IDateTime.R#L275

    xlt <- as.POSIXlt(x)

To 

    xlt <- as.POSIXlt(x, tz = 'UTC')

(that's what is seems is done by `lubridate::isoweek` -> `lubridate::make_datetime`):

```
# lubridate::isoweek
#  c/f https://github.com/tidyverse/lubridate/blob/e4ad2c3024c4e114c5f046c7fa0d73e2168f0d2b/R/accessors-week.r#L36
#  ** on CRAN, lubridate::isoweek is what now corresponds to
#      lubridate:::.other_week(x, 1)
x <- as.POSIXlt(x)
```

```
# lubridate::make_datetime
#   c/f https://github.com/tidyverse/lubridate/blob/c10c8cfff6a8d86e58c5865c1fb1c93463be647e/R/instants.r#L109
.POSIXct(numeric(), tz = tz) # default tz = 'UTC'
```

(actually, on further look, these look the same. leaving this for my own notes)

**Comment by SimonCoulombe** [comment]
data.table::isoweek("2014-12-29")
[1] 1
> data.table::isoweek(as.Date("2014-12-29"))
[1] 0

**Comment by MichaelChirico** [comment]
@SimonCoulombe thanks! I've spotted the problem and will push a fix soon.
